### ISSUE TITLE
VanillaExtract: classNames prop does not take precedence over component base styles unlike styles prop

### ISSUE THREAD

**Issue by kevinsmouts** [mantinedev/mantine#4886]
### What package has an issue

@mantine/core

### Describe the bug

On Mantine v7, from my understanding of the Styles API documentation two functions are equivalent:

```
<Text styles={{ root: { fontSize: 12 } }} >
   {content}
 </Text>
```
is equivalent to
```
//css
.textSize{font-size: 12px;}
// page
<Text classNames={{ root: classes.textSize }} >
   {content}
 </Text>
```
as you can see here: https://codesandbox.io/s/styles-precedence-j3pcsk?file=/src/App.tsx

Well, with Vanilla Extract it doesn't work:
![image](https://github.com/mantinedev/mantine/assets/1415547/4413f98e-831e-4ee7-b945-84356faf59a8)

### What version of @mantine/* packages do you have in package.json? (Note that all @mantine/* packages must have the same version in order to work correctly)

7.0.0

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

**Comment by rtivital** [comment]
The issue is resolved in 7.1.0, you can now use CSS layers to control the order of styles – https://mantine.dev/styles/mantine-styles/#css-layers

We cannot do anything else on our side because Mantine provides static css and cannot control how this file is injected by third-party tools. If you think that the issue is not resolved for you, please report it in the issues of the tools you are using.

**Comment by anri-asaturov** [comment]
For anyone still struggling with this and not wanting to use layers:

Just make sure to import mantine styles before any of the react component that import any of the vanilla-extract files.

Move `import '@mantine/core/styles.css'` and the rest of mantine style imports to the very top of the file that renders react root. The bundler (vite in my case) has to see them first.
